pkg/controllers/workgenerator/controller.go

```diff
@@ -234,18 +249,13 @@ func (r *Reconciler) updateBindingStatusWithRetry(ctx context.Context, resourceB
 			}
 			// Work generator is the only controller that updates conditions excluding rollout started which is updated by rollout controller.
 			if rolloutCond := latestBinding.GetCondition(string(fleetv1beta1.ResourceBindingRolloutStarted)); rolloutCond != nil {
-				found := false
```

Comment:
now that we check the rolloutCondition before, this is now a real "unexpected" behavior then?